### ISSUE TITLE
DESIGN: layered anchor + segmentation model (link anything → any page segment; build on ALTO/PageXML/TEI)

### ISSUE THREAD

**Issue by dtubb** [dtubb/fichero#4635]
**Design-first, research-heavy spec** (constitution: design issue first; docs/contributor/TESTING-CONSTITUTION.md). Creative director's brief, 2026-09-08. NOT a one-off — this is the foundation the KG source-anchoring (F5, #4623) grows into.

## The question
How does anything meaningful (claim, entity, transcription, POS tag, note, version, chain-of-thought) point INTO a page — at any granularity and any layer — extensibly?

## Decisions (creative director)
- **What anchors:** GENERAL — anything → any segment (one mechanism for claims, entities, transcriptions, POS, notes, metadata, versions/history).
- **Granularity:** OPEN/LAYERED — geometry (box OR polygon + angle/direction) + a granularity level + a semantic-layer tag; levels/layers are an extensible vocabulary (no schema break to add one).
- **Standards:** build ON + import/export TO **ALTO / PageXML / TEI** (and review **ZOLO-text**); but go beyond them where it helps under-served scripts/languages. Investigate whether they capture everything first.

## Granularity hierarchy (draft, to refine)
Strokes → Characters → Words → Baselines → Regions / Gloss / Marginalia (top/bottom/side) / Interline / Forms / Parts-of-Page / Signature → Page-Split → Pages → Multiple Pages → Maps. Plus: drawing on page; EXCLUDE part of a page/document; trace around an image; regions via SVG(?); across multiple images(?).

## Must handle
- BiDi / RTL / Top-to-Bottom scripts (Kraken supports — see below); Arabic, ancient languages, cuneiform.
- Language AND script tracked separately (per segment). Language authority: https://glottolog.org
- Images (not just text): trace/segment images; maps; art(?).
- Versions / history / chain-of-thought on anchors + transcriptions.

## Research to do (this is a spec to properly investigate)
- Review the actual **ALTO, PageXML, TEI** specs — do they capture strokes? maps? page-split? exclusions? multi-image? Where do they fall short for our scripts?
- **ZOLO-text** spec.
- **Kraken 7.1**: https://kraken.re/7.1/index.html — what does it give us (RTL/BiDi/TTB, ALTO/PageXML/abbyyXML/hOCR output, word bbox + char cuts, trainable layout/reading-order)?
- **badianeai/AnandaSky**: https://huggingface.co/badianeai/AnandaSky
- **MEI** (Music Encoding Initiative) — music notation anchoring; are there art/other domain encodings worth aligning with?
- Survey the wider literature (this is a well-studied area).

## Deliverable
`docs/contributor/specs/anchor-model.md` — the layered anchor design + a standards-mapping table (our layers ↔ ALTO/PageXML/TEI/MEI) + import/export plan + extension rules. Then design-led implementation waves (F5 one-builder is wave 0).

**Comment by dtubb** [comment]
More creative-director design context (2026-09-08) — the anchor/segmentation model must be **defaults-driven and language/script-aware**:

- Provide **defaults a user can choose**, depending on their **project / language / script / document type / where there are language challenges**. (Relates to existing #2116 "Model selection that EDUCATES + evaluates: per-language/script fit".)
- Design a **Taxonomy of Language Challenges**: the Language Challenges themselves; overlap in tasks across groups; the technical challenges we face; and that people with those skills understand them. Examples to research: **mixing Korean + Chinese**, and others TBD. → This is a research project: survey what the real script/language challenges are.
- Language authority + a worked reference: https://glottolog.org and https://glottolog.org/static/nebrija.jpg

So the model + the local-runtime/model defaults are coupled: which runtime/model/segmenter to default to is a function of language + script + document type + the known challenges for that combination. Fold a "defaults by language/script/doc-type" section into the anchor-model spec (or a sibling spec), grounded in a language-challenge taxonomy.

**Comment by dtubb** [comment]
## More design input (creative director, 2026-09-08) — direction, Kraken controls, granularity

### Text direction (per segment, not per doc)
Segments carry a **reading direction** (from Kraken's `--text-direction`):
- `horizontal-lr` — Latin; mixed LTR/RTL with principal LTR
- `horizontal-rl` — Arabic; mixed LTR/RTL with principal RTL
- `vertical-lr` — vertical, read left→right
- `vertical-rl` — vertical, read right→left
Plus arbitrary **polygons** + **angle/direction** per segment. A page can mix directions (Latin body with RTL/vertical interlines).

### Kraken segmentation controls we want to expose/store (kraken.re)
`--scale` (avg line height), `--maxcolseps` (max columns; 0 = uni-column), black/white `--colseps`, `--remove-hlines` (helps some Arabic), `--pad`, `--text-direction`, `--mask` (a B/W mask suppressing page areas from line detection — i.e. store an EXCLUSION mask as a segment/region). All of Kraken's layout knobs should be capturable per document/run.

### Granularity hierarchy (refined)
Strokes → Characters → Words → Baselines → Regions / **Gloss** / **Marginalia** / **Interline** / Forms / Parts-of-Page / **Signature** / **end-of-line** → Page-Split → Pages → Multiple Pages → Maps(?). Plus: **drawing on page**; **EXCLUDE part of a page/document** (the mask); trace around an image.

### Open question folded in
Fonts: do we store a font per segment / per script-area / per language? A page may have 5 languages, or Latin body with interlined another language. See the companion language/script/coverage design (filed separately).

**Comment by dtubb** [comment]
Architecture decisions (creative director, 2026-09-08): (1) a page-segment is a **first-class 'Segment' record** (stroke/char/word/line/region/… with geometry + direction + language + script) that claims/entities/transcriptions/annotations REFERENCE by id — segments exist independently (e.g. from Kraken layout) before anything anchors to them; one segment, many referrers. (2) Storage = **native rich model in the DB + map to ALTO/PageXML/TEI/hOCR on import/export** — full fidelity for things the standards can't hold (strokes, coverage tiers, exclusion masks) plus interop via mappers.

**Comment by dtubb** [comment]
## Refinement: slots are VERSIONED COLLECTIONS at the segment grain (creative director, 2026-09-08)

A segment doesn't hold one transcription or one representation — it holds **many, each versioned**:

```
Segment (one anchor / region on a page)
  ├─ Transcription[]   each { text, language, script, edition, version, provenance }
  ├─ Representation[]  each { kind: image|text|tokens|image-vec|text-vec, version, provenance }
  ├─ language/script   segment DEFAULT (a transcription may override)
  └─ export map
```

Consequences:
- **Language/script live on the Transcription, not only the Segment.** The same region can be rendered in different scripts — a bilingual gloss = 2 transcriptions on 1 segment; graphemic vs normalized = 2 transcriptions; a Latin line with a Greek insertion may split or carry mixed script. Segment keeps a default; each transcription can differ.
- **Representations are versioned, append-only.** Re-crop the image, or embed with a new model → a NEW representation version; the old one is retained (provenance/version slot at finest grain — never overwrite).
- **Profile seeds the defaults.** Picking a profile decides which transcriptions/representations a document gets by default (medieval → diplomatic + normalized transcriptions, image crop + HTR-model text-vector). Profile endorsed as the hinge.

This is the "editions/versions" thread (#4638) landing at the segment grain, and it's the read-model the segment-representations vertical slice will expose across backend → inspector → MCP/CLI → tests → export.

**Comment by dtubb** [comment]
## Surface = the Preview (source) pane; anchors are spatial AND temporal (creative director, 2026-09-08)

### Where segments live in the UI
Segments/representations are shown and **edited on the actual media in the Preview (source) pane** — the polygon drawn on the image/PDF, editable in place — NOT only in the inspector. The inspector shows a segment's versions/metadata + plausible readings; the Preview pane is where you see the anchor on the source and adjust it. Surface home: **Preview View - Image Editing (#168)** (has the bbox/region-drawing work already); inspector side stays in Inspector View - Knowledge (#151).

### Anchor = any addressable slice of any medium (W3C Media Fragments)
image/PDF is priority, but the source pane also holds audio, video, other formats — so the anchor generalizes:
- **spatial** (image/PDF): a polygon / `#xywh=x,y,w,h`
- **temporal** (audio/video): a time span / `#t=start,end`
- **both** (video): region + time
This is exactly **W3C Media Fragments** — one model addresses image regions, PDF regions, audio clips, video shots. A 'segment' is any such slice; the same four slots (anchor / language+script / provenance+version / export map) hang off all of them.

### Consequence
- Ship **image/PDF** first (spatial polygons on the Preview pane); **audio/video reuse the identical Segment model** with a temporal anchor — no new concept, just a second anchor kind.
- Ties W3C Web Annotation (already an archival-format target) — a Media-Fragment selector IS the anchor serialization for export.

**Comment by dtubb** [comment]
## Technique: VLM-emitted SVG as segment-geometry capture (creative director, 2026-09-08)

Some VLMs are genuinely good at **outputting SVG**. Use that to **capture segments**: ask the VLM to describe a page's regions/lines/polygons **as SVG paths**, and that SVG becomes the **anchor geometry** directly — no bespoke coordinate JSON.

Why it fits:
- SVG paths/polygons map straight onto the segment anchor (arbitrary polygons, not just bboxes). **W3C Web Annotation defines an SVG selector** — so VLM SVG IS a standards-valid anchor serialization (export slot for free).
- **Training loop** (ties #4642): (page-image → SVG-geometry) pairs are training data for a segmentation model; "good luck last year" with this — and it supports **iterative improvement**: VLM proposes SVG → human corrects on the Preview pane → corrected (image → SVG) feeds the next model, which proposes better SVG. Same provenance/version + reasoning capture (#4636) applies to the geometry, not just the text.
- Pairs with the Kraken/DLA segmenters as an ALTERNATE segment source — the profile (#4639) can pick VLM-SVG vs Kraken per document type.

CAPTURE only; design after KG CRUD + bugs. Anchor-geometry source option for #4635; training/iterative-improvement rides #4642.

**Comment by dtubb** [comment]
## Missing slot: EVALUATION / error rates on regions/segments/transcriptions (creative director, 2026-09-08)

Regions (and segments, transcriptions, zones) need **error rates / quality metrics** — a measurement slot alongside anchor / language-script / provenance / export:

- **transcription**: CER / WER (character/word error rate) vs a reference
- **segmentation/region**: IoU, precision/recall, panoptic quality vs ground truth
- **general**: model confidence, agreement across model versions, human-review status

Why it matters:
- It's how you **measure the flywheel** (#4642): which examples the current model gets WRONG → prioritize those for human review; did the fine-tuned student actually beat the teacher/prior version?
- Metrics attach per VERSION (so you can compare v1/v2/v3 of a transcription or a segmentation), riding the provenance/version model (#4636).
- Eval tooling already named in this program: **KaMI** (HTR eval), CATMuS guidelines, Pycocotools (COCO metrics for layout). These compute the metrics; Fichero stores + shows them.
- Open: what else belongs here — inter-annotator agreement? calibration? per-glyph confusion matrices? (creative director: "not sure what else").

Add an **evaluation slot** to the model; surface it in the Reader/inspector (how good is this reading?) and export it (JSONL metrics per record, #4640). CAPTURE only; design after KG CRUD + bugs.

**Comment by dtubb** [comment]
## Consequence: embeddings move from CONTENT-level to SEGMENT-level (creative director, 2026-09-08)

The segment model reshapes embeddings. Today embeddings are computed at the **content** (document/page) grain. With segments, each **segment** carries its own embedding — the **image-vector / text-vector representation** already in the model — so:

- Semantic search operates at **segment resolution** (find the line/region, not just the document) — hits land on an anchor with a highlight, not a whole page.
- The embedding IS a representation version (append-only; a new embedding model = a new version) with its own provenance (#4636) and eval (#4635 metrics).
- Coarser embeddings (page/document) become **aggregations** of segment embeddings, not the primary unit.
- loove/coverage informs which embedding model per language/script (#4637).

This is a change to the embeddings subsystem (Search View - Engine #186, embeddings-deep-dive program): index at the segment grain, keyed by segment id, versioned by model. CAPTURE only; design after KG CRUD + bugs. Ties segment-representations spec (image-vector/text-vector reps).

**Comment by dtubb** [comment]
## Reading order is EDITORIAL (and therefore versioned) (creative director, 2026-09-08)

Segments/zones have a **reading order** — but it's not an objective fact, it's an **editorial choice**. Two editors can order the same page differently: columns L-then-R vs interleaved, where marginalia/interlinear glosses fall in the sequence, how a table or a torn page is linearized.

So reading order is NOT a single field on the anchor — it's an **edition-level decision** (ties #4638 editions), carrying the same slots as everything else:
- it belongs to an **edition/version** (diplomatic order vs a normalized reading order can differ),
- with **provenance** (who/which model ordered it) + **reasoning** (#4636) — a model proposes an order, a human overrides it, both recorded,
- and its own **export** (TEI reading order / METS structMap / IIIF ranges).

Model: an **Edition** owns an **ordered sequence** of segment/zone references (the reading order); the segment keeps a default index, but the authoritative order lives in the edition and is versioned/attributable. A different edition = a different order over the same segments. CAPTURE only; design after KG CRUD + bugs. Ties #4638 (editions), #4636 (provenance/versions), zones (#4639).
